### PR TITLE
Feature/text field mask

### PR DESCRIPTION
**Pull request by fdiassilva**
- Listener click in right icon
- Color change to right icon

## Useful links
- [Contributing](https://github.com/mercadolibre/fury_andesui-android/blob/develop/CONTRIBUTING.md) has more information and tips for a great pull request. This will give you a nice introduction about how to contribute with Andes.
- [Wiki](https://github.com/mercadolibre/fury_andesui-android/wiki) has all the info you need to start with Andes. Even I heard there's a video there...

## Motivation
Tell us why you are here and why you took time to develop new code for our beloved lib.

## Description
Please describe your intentions with this new code you want to introduce inside Andes UI lib.
Try to answer those questions:
- What is it about?
Agregué la posibilidad de agregar una máscara al campo AndesTextfield

- How does it work?

## Affected component
AndesTextfield

## Frontify component link
The UX team behind Andes keeps all the Andes Components inside the Frontify site. Each component has its own section. Let us know the section of the component you want to modify.

## Screenshots
mask_andes

## Checks
Are you sure you followed all those steps? In such case, let's say with me "I solemnly declare that ...":
- [ ] I have coded an example inside the Demo App,
- [ ] I have added proper tests,
- [x] I have modified the [Changelog](https://github.com/mercadolibre/fury_andesui-android/blob/develop/CHANGELOG.md) file,
- [ ] I have updated GitHub wiki,
- [x] I have the explicit approval of one or more members of the UX Team,
- [ ] I have updated the version in gradle.properties file.

## Change type
This is easy, let us know what this code is about:
- [ ] This code adds a new component
- [x] This code includes improvements to an existent component
- [ ] This code improves or modifies ONLY the Demo App.

## Check common errors
Whether you are an Android or iOS developer and if you're still there then we'll give you a present:
https://proandroiddev.com/how-to-maximize-androids-ui-reusability-5-common-mistakes-cb2571216a9f

### CODE REVIEW

**Comment by furyGorgoryAdmin** [comment]
### Informative Menu <h2>  ### 

### Use any of these tags in a pull request description or leave a comment ### 
| Tag | Description |
|---|---|
| **[canary]**| Creates a build that can only be deployed with canary strategy. | 
| **[build]** |  Creates version or build from a branch that has been tagged as "no releasable". | 
| **[name:version-name]** |Specify the name of the build. For example: [build] [name:1.2.3-name] or [canary] [name:1.2.3-name].|
| **[major]** | To automatically increase the first position of the version. For example: [build] [major] or [canary] [major].|
| **[minor]** | To automatically increase the second position of the version. For example: [build] [minor] or [canary] [minor]. |
| **[patch]** | To automatically increase the third position of the version. For example: [build] [patch] or [canary] [patch].|  

[For more information about ci-cd](http://docs.furycloud.io/?lang=esp&section=businessassurance)
[For more information about semver](https://semver.org/#summary)

File: components/src/main/java/com/mercadolibre/android/andesui/textfield/AndesTextfield.kt [hunk 1]
```diff
@@ -371,11 +376,23 @@ class AndesTextfield : ConstraintLayout {
...
+
+                val textWithoutMask = cleanText(charSequence.toString())
+
+                counterComponent.text = resources.getString(R.string.andes_textfield_counter_text,
+                    textWithoutMask?.length, config.counterLength)
             }
         })
     }
 
+    private fun cleanText(text: String): String? {
```
Comment (RodriP) [inline components/src/main/java/com/mercadolibre/android/andesui/textfield/AndesTextfield.kt:388]: le sumamos javadoc?

File: components/src/main/java/com/mercadolibre/android/andesui/textfield/AndesTextfield.kt [hunk 2]
```diff
@@ -484,14 +501,24 @@ class AndesTextfield : ConstraintLayout {
     /**
      * Set the right content to icon and provides an interface to give the icon path.
      */
-    fun setRightIcon(iconPath: String) {
+    fun setRightIcon(iconPath: String, listener: OnClickListener? = null, colorIcon: Int? = R.color.andes_gray_800) {
         rightContent = AndesTextfieldRightContent.ICON
         val rightIcon: SimpleDraweeView = rightComponent.getChildAt(0) as SimpleDraweeView
+
+        var color: AndesColor? = null
+        if (colorIcon != null) {
```
Comment (RodriP) [inline components/src/main/java/com/mercadolibre/android/andesui/textfield/AndesTextfield.kt:509]: que pasa si es null?

File: components/src/main/java/com/mercadolibre/android/andesui/textfield/AndesTextfield.kt [hunk 3]
```diff
@@ -525,6 +552,34 @@ class AndesTextfield : ConstraintLayout {
...
+     * @param mask mask pattern ##-###-######-# digits sample 0123456789-
+     */
+    fun setTextFieldMask(mask: TextFieldMask, listener: TextFieldMaskWatcher.OnTextChange?) {
+
+        if (maskWatcher == null) {
+            maskWatcher = TextFieldMaskWatcher(mask.mask, listener)
+            textComponent.addTextChangedListener(maskWatcher)
+        }
+
+        maskWatcher?.setMask(mask.mask)
```
Comment (RodriP) [inline components/src/main/java/com/mercadolibre/android/andesui/textfield/AndesTextfield.kt:566]: es necesario el ? si lo estamos inicializando arriba si es null?

File: components/src/main/java/com/mercadolibre/android/andesui/textfield/maskTextField/TextFieldMaskWatcher.kt [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            textWithoutMask = textWithoutMask.replace(char.toString(), "")
+        }
+        return textWithoutMask
+    }
+
+    fun getMaxLength(): Int {
+        return if (mask.isNotEmpty()) {
+            mask.length - mask.replace("#", "").length
+        } else {
+            0
```
Comment (RodriP) [inline components/src/main/java/com/mercadolibre/android/andesui/textfield/maskTextField/TextFieldMaskWatcher.kt:53]: lo sumamos como constante por default?

File: components/src/main/java/com/mercadolibre/android/andesui/textfield/maskTextField/TextFieldMaskWatcher.kt [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+        isRunning = false
+    }
+
+    fun setMask(mask: String) {
+        this.mask = mask
+    }
+
+    fun cleanMask(text: String): String {
+        val maskChars = mask.replace("#", "")
```
Comment (RodriP) [inline components/src/main/java/com/mercadolibre/android/andesui/textfield/maskTextField/TextFieldMaskWatcher.kt:41]: lo usamos en varios lados, podemos agregarlo como constante?"#", ""

Comment (fdiassilva) [reply components/src/main/java/com/mercadolibre/android/andesui/textfield/maskTextField/TextFieldMaskWatcher.kt]: done!

File: components/src/main/java/com/mercadolibre/android/andesui/textfield/maskTextField/TextFieldMaskWatcher.kt (hunk 4 above)
Comment (fdiassilva) [reply components/src/main/java/com/mercadolibre/android/andesui/textfield/maskTextField/TextFieldMaskWatcher.kt]: done!

File: components/src/main/java/com/mercadolibre/android/andesui/textfield/AndesTextfield.kt (hunk 3 above)
Comment (fdiassilva) [reply components/src/main/java/com/mercadolibre/android/andesui/textfield/AndesTextfield.kt]: done!

**Pull request merged**